__tests__/todo-config-test.ts

```diff
@@ -0,0 +1,244 @@
+      setupEnvVar('TODO_DAYS_TO_WARN', '1');
+      setupEnvVar('TODO_DAYS_TO_ERROR', '2');
+
+      const config = getTodoConfig(project.baseDir, {
+        warn: 5,
+        error: 10,
+      });
+
+      expect(config).toEqual({
+        warn: 5,
+        error: 10,
+      });
+    });
+
+    it('can override defaults with null values', () => {
+      project.writeTodoConfig({
+        warn: 1,
+        error: 2,
+      });
+
+      const config = getTodoConfig(project.baseDir, {
+        warn: undefined,
+        error: undefined,
+      });
+
+      expect(config).toEqual({
+        warn: undefined,
+        error: undefined,
+      });
+    });
+
+    it('can override defaults with single null value', () => {
+      project.writeTodoConfig({
+        warn: 1,
+        error: 2,
+      });
+
+      const config = getTodoConfig(project.baseDir, {
+        error: undefined,
+      });
+
+      expect(config).toEqual({
+        warn: 1,
+        error: undefined,
+      });
+    });
+
+    it('throws if warn is equal to error', () => {
+      expect(() => getTodoConfig(project.baseDir, { warn: 5, error: 5 })).toThrow(
+        'The provided TODO configuration contains invalid values. The `warn` value (5) must be less than the `error` value (5).'
+      );
+    });
+
+    it('throws if warn is greater than to error', () => {
+      expect(() => getTodoConfig(project.baseDir, { warn: 10, error: 5 })).toThrow(
+        'The provided TODO configuration contains invalid values. The `warn` value (10) must be less than the `error` value (5).'
+      );
+    });
+  });
+
+  describe('writeTodoConfig', () => {
+    it('does not change package.json if lintTodo config present', () => {
+      project.writeTodoConfig({
+        warn: 1,
+        error: 2,
+      });
+
+      const originalPkg = readFileSync(join(project.baseDir, 'package.json'), { encoding: 'utf8' });
+
+      writeTodoConfig(project.baseDir, { warn: 10, error: 20 });
+
+      const pkg = readFileSync(join(project.baseDir, 'package.json'), { encoding: 'utf8' });
+
+      expect(pkg).toEqual(originalPkg);
+    });
+
+    it('adds the todo config when one does not exist', () => {
+      writeTodoConfig(project.baseDir, { warn: 1, error: 2 });
+
+      const pkg = readFileSync(join(project.baseDir, 'package.json'), { encoding: 'utf8' });
+
+      expect(pkg).toMatchInlineSnapshot(`
+        "{
+          \\"name\\": \\"fake-project\\",
+          \\"version\\": \\"0.0.0\\",
+          \\"keywords\\": [],
+          \\"license\\": \\"MIT\\",
+          \\"description\\": \\"Fake project\\",
+          \\"repository\\": \\"http://fakerepo.com\\",
+          \\"dependencies\\": {},
+          \\"devDependencies\\": {},
+          \\"lintTodo\\": {
+            \\"decayDays\\": {
+              \\"warn\\": 1,
+              \\"error\\": 2
+            }
+          }
+        }"
+      `);
+    });
+
+    it('adds the todo config when one does not exist respecting trailing whitespace', () => {
```

Comment:
Does the other test check the other side of this (when a trailing whitespace is not present)? I kind of expect two tests here: one for when trailing whitespace exists (confirming it is preserved), and another when trailing whitespace does not exist (confirming it is not added).

Reply:
Correct. The other test before this one checks if its not present.